instrumentation/src/androidTest/java/com/bumptech/glide/DarkModeTest.java

```diff
@@ -51,6 +52,45 @@ public void before() {
     assumeTrue(VERSION.SDK_INT >= VERSION_CODES.Q);
   }
 
+  @Test
+  public void load_withDarkModeActivity_vectorDrawable_usesDarkModeColor() {
+    try (ActivityScenario<FragmentActivity> scenario = darkModeActivity()) {
+      scenario.onActivity(
+          activity -> {
+            ViewGroup container = findContainer(activity);
+            ImageView imageView = newFixedSizeImageView(activity);
+            container.addView(imageView);
+
+            Glide.with(activity)
+                .load(R.drawable.vector_drawable)
+                .override(Target.SIZE_ORIGINAL)
+                .into(imageView);
+          });
+
+      onIdle();
+      scenario.onActivity(
+          activity -> {
+            ViewGroup container = findContainer(activity);
+            ImageView imageView = (ImageView) container.getChildAt(0);
+            Bitmap result = drawToBitmap(imageView.getDrawable());
+            Bitmap expected = drawToBitmap(activity.getDrawable(R.drawable.vector_drawable_dark));
+            assertThat(result).sameAs(expected);
+          });
+    }
+  }
+
+  private static Bitmap drawToBitmap(Drawable drawable) {
+    int width = drawable.getIntrinsicWidth();
+    int height = drawable.getIntrinsicHeight();
+
+    Bitmap result = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888);
+    Canvas canvas = new Canvas(result);
+    drawable.setBounds(0, 0, width, height);
+    drawable.draw(canvas);
+    canvas.setBitmap(null);
+    return result;
+  }
```

Comment:
**Performance Issue**: The method 'drawToBitmap' creates a new Bitmap and Canvas every time it's called. This can be highly inefficient and memory-intensive, especially for large numbers of drawables or high-resolution images. Reusing bitmaps or using a bitmap pool could significantly improve memory usage and performance. <br> **Fix**: Consider using a Bitmap pool to reuse Bitmap objects rather than creating a new one each time. Glide provides a BitmapPool interface for this purpose. Alternatively, if the drawable sizes are known and fixed, preallocate a single Bitmap and Canvas and reuse them for each drawable. <br> **Code Suggestion**: 
 ```
 +  private static Bitmap reusableBitmap;
 +  private static Canvas reusableCanvas;

    private static Bitmap drawToBitmap(Drawable drawable) {
      int width = drawable.getIntrinsicWidth();
      int height = drawable.getIntrinsicHeight();

 +    if (reusableBitmap == null || reusableBitmap.getWidth() != width || reusableBitmap.getHeight() != height) {
 +      reusableBitmap = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888);
 +      reusableCanvas = new Canvas(reusableBitmap);
 +    }
 +    reusableCanvas.setBitmap(reusableBitmap);
      drawable.setBounds(0, 0, width, height);
      drawable.draw(reusableCanvas);
 +    reusableCanvas.setBitmap(null);
      return reusableBitmap;
    }
 ```